File: NSpace/Event.cs
//----------------------------------------
// Copyright (c) 2010, Dmitry Zamkov 
// Open source under the new BSD License
//----------------------------------------
using System;
using System.Collections.Generic;

namespace NSpace
{
    /// <summary>
    /// A vector paired with a time to represent a point in spacetime.
    /// </summary>
    public struct Event
    {
        public Event(Vector Point, Time Time)
        {
            this.Point = Point;
            this.Time = Time;
        }

        public Vector Point;
        public Time Time;
    }
}

File: NSpace/src/NSpace/Event.hs
-----------------------------------------------------------------------------
--
-- Module      :  NSpace.Event
-- Copyright   :  (c) 2010, Dmitry Zamkov
-- License     :  BSD3 (See LICENSE)
--
-----------------------------------------------------------------------------

module NSpace.Event (
	Event(..),
	Time
) where

import NSpace.Vector

-- Represents a time within a frame of reference

type Time	=	Double

-- An event is a point in four-dimensional space time, specified with an 
-- actual position and the time the point is at that position.

data Event	=	Event {
						getPosition	::	Vector,
						getTime		::	Time } deriving (Eq, Show)